taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -13,7 +17,113 @@
 from ..build_config import get_components
 
 
+logger = logging.getLogger(__name__)
+
+ALL_COMPONENTS = object()
+# We always build components that have changed files or dependencies, but we also need
+# to rebuild some or all components when other files change. This structure
+# maintains the latter mapping.
+FILE_PATTERNS_TO_AFFECTED_COMPONENTS = {
```

Comment:
This would be better as a top-level key in the `kind.yml`.

Reply:
Good idea.